standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -5297,6 +5303,28 @@ is performed on that db (e.g. show tables, created table, etc).
             return response;
           }
         }
+
+        if (checkForConcurrentCtas && isValidTxn(txnId)) {
```

Comment:
embed this check with above zeroWaitReadEnabled
````
if ((zeroWaitReadEnabled || isExclusiveCTAS) && isValidTxn(txnId)) {
....
if (lockType == LockType.SHARED_READ || isExclusiveCTAS) {
....
````

Reply:
done